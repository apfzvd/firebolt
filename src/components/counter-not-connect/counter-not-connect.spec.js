import { shallow } from 'enzyme'
import React from 'react'
import Counter from './index'

let wrapper

describe('not connect', () => {
  beforeEach(() => {
    wrapper = shallow(<Counter />)
  })

  it('snapshot counter not connect', () => {
    expect(wrapper.debug()).toMatchSnapshot()
  })

  it('renders with initial value of 0', () => {
    expect(wrapper.find('.value').text()).toEqual('0')
  })

  it('increments value when clicking +1', () => {
    wrapper.find('#increment-btn').simulate('click')
    expect(wrapper.find('.value').text()).toEqual('1')
  })

  it('decrement value when clicking -1', () => {
    wrapper.find('#decrement-btn').simulate('click')
    expect(wrapper.find('.value').text()).toEqual('-1')
  })

  it('reset value for 0', () => {
    wrapper.find('#reset-btn').simulate('click')
    expect(wrapper.find('.value').text()).toEqual('0')
  })
})
