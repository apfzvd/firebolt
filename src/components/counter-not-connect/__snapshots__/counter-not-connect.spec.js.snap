// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not connect snapshot counter not connect 1`] = `
"<div className={[undefined]}>
  <span>
    Componente n√£o conectado -
     
    <strong>
      Valor:
      <span className=\\"value\\">
        0
      </span>
    </strong>
  </span>
  <div>
    <Button id=\\"increment-btn\\" className={[undefined]} onClick={[Function: handleIncrement]} disabled={false} color=\\"primary\\" size=\\"medium\\" type=\\"button\\" active={false}>
      Aumentar valor
    </Button>
    <Button id=\\"decrement-btn\\" onClick={[Function: handleDecrement]} className={[undefined]} disabled={false} color=\\"primary\\" size=\\"medium\\" type=\\"button\\" active={false}>
      Diminuir valor
    </Button>
    <Button id=\\"reset-btn\\" onClick={[Function: handleReset]} className={[undefined]} disabled={false} color=\\"primary\\" size=\\"medium\\" type=\\"button\\" active={false}>
      Zerar
    </Button>
  </div>
</div>"
`;
